### ISSUE TITLE
Provider logs error with api key on hostname lookup failure

### ISSUE THREAD

**Issue by freakinhippie** [fortinetdev/terraform-provider-fortios#144]
```
Firmware: FortiOS v6.4.4 build1803 (GA)
Provider: fortinet/fortios/1.10.3
Terraform: 0.14.7
```

When the hostname provided to the `fortios` provider fails to resolve in DNS, the provider returns an error which includes "`access_token=[API_KEY]`". The api key value should be sanitized before logging to prevent credential leakage.

While this example is contrived, my actual use case was uncovered due to a brief lag in route53 making the DNS record available in my environment.

Example:

_providers.tf_:

```hcl
# Configure the FortiOS Provider for FortiGate
provider "fortios" {
  alias        = "primary"
  hostname     = "fakehost.example.com"
  token        = "Gyy8fprxd9c5yH85Nf91phq7scjHkt" # for example purposes, not a real token
  insecure     = "false"
  cabundlefile = "${path.module}/data/cert_chain.crt"
}
```

**output:**

```sh
Error: Error create fortios client: connection error: cannot send request, # lines wrapped for clarity
  lost connection to firewall with error:
  Get "https://fakehost.example.com/api/v2/cmdb/system/global?access_token=Gyy8fprxd9c5yH85Nf91phq7scjHkt":
  dial tcp: lookup fakehost.example.com on 127.0.0.53:53: no such host

  on providers.tf line 2, in provider "fortios":
   2: provider "fortios" {
```

**Comment by frankshen01** [comment]
Thanks to @freakinhippie for the issue. It has been fixed and released in version 1.10.4 of the Terraform FortiOS provider. Please see: https://registry.terraform.io/providers/fortinetdev/fortios/latest . Thank you!

**Comment by freakinhippie** [comment]
Thank you, @frankshen01. I've confirmed it is fixed in 0.10.4.
